src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SMKRuleReference.kt

```diff
@@ -23,5 +22,25 @@ class SMKRuleReference(node: ASTNode): PyElementImpl(node), PsiNamedElement {
         return this
     }
 
+    override fun getReference(): PsiReference? = SMKRulePSIReference(this, TextRange(0, textLength))
+
     private fun getNameNode() = getIdentifierNode(node)
+}
+
+class SMKRulePSIReference(
```

Comment:
Psi element is called `SMKRuleReference` and reference called  `SMKRule`**PSI**`Reference` - it is very not intuitive. Sounds like `SMKRuleReference` is some reference and `SMKRulePSIReference` is some psi element.

I suggest using `SmkReferenceExpression` for psi elment (like `PyReferenceExpression`) and `SmkRuleOrCheckpointNameReference` for reference.